src/rime/core_module.cc

```diff
@@ -20,13 +20,27 @@ static void rime_core_initialize() {
   LOG(INFO) << "registering core components.";
   Registry& r = Registry::instance();
 
-  auto config = new ConfigComponent;
-  config->InstallPlugin(new AutoPatchConfigPlugin);
-  config->InstallPlugin(new DefaultConfigPlugin);
-  config->InstallPlugin(new LegacyPresetConfigPlugin);
-  config->InstallPlugin(new LegacyDictionaryConfigPlugin);
-  r.Register("config", config);
-  r.Register("schema", new SchemaComponent(config));
+  const ResourceType build_output{"config", "build/", ".yaml"};
+  auto config_builder = new ConfigComponent<ConfigBuilder>(
+      [&](ConfigBuilder* builder) {
+        builder->InstallPlugin(new AutoPatchConfigPlugin);
+        builder->InstallPlugin(new DefaultConfigPlugin);
+        builder->InstallPlugin(new LegacyPresetConfigPlugin);
+        builder->InstallPlugin(new LegacyDictionaryConfigPlugin);
+        builder->InstallPlugin(new BuildInfoPlugin);
+        builder->InstallPlugin(new SaveOutputPlugin(build_output));
+      });
+  r.Register("config_builder", config_builder);
+
+  auto config_loader = new ConfigComponent<ConfigLoader>(build_output);
+  r.Register("config", config_loader);
```

Comment:
it's no longer possible to access `user.yaml` and `installation.yaml` via the `RimeConfigOpen` API. is there a workaround?

Reply:
Err...

Reply:
Added 'RimeUserConfigOpen()'